app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,53 @@
+class ReviewsController < ApplicationController
+
+  def index
+    render json: Review.all
+  end
+
+  def new
+    @review = Review.new
+  end
+
+  def create
+    @review = Review.new(review_params)
+    if @review.valid?
+      @review.save
+
+      update_restaurant_rating(@review.restaurant_id)
+      render json: @review
+    else
+      render json: @review.errors, status: :bad_request
+    end
+  end
+
+  def destroy
+    Review.destroy(params[:id])
+  end
+
+  def update
+    rest = Review.find(params[:id])
+    rest.update_attributes(update_review_params)
+    if params[:rating].present?
+      update_restaurant_rating(rest.restaurant_id)
+    end
+
+    render json: rest
+  end
+
+  private
+
+  def review_params
+    params.permit(:id, :restaurant_id, :reviewer_name, :rating, :comments)
+  end
+
+  def update_review_params
+    params.permit(:rating, :comments)
+  end
+
+  def update_restaurant_rating(rest_id)
```

Comment:
This is something I'd extract to a class outside of the controller, as it's doing calculations that aren't in the context of routing or handling responses.
These chunks of code usually expand during the life of the project and make the Controller cumbersome and hard to maintain.